### PR TITLE
Bugfix/deseng227 Add doc blocks to all "service" classes

### PR DESCRIPTION
**Pull request by Baelx**
PR 1 of 2.

Also, there were breaking changes to the "input-modal" component from a previous PR that have been corrected here.

### CODE REVIEW

File: src/app/services/api.ts [hunk 1]
```diff
@@ -94,15 +111,27 @@ export class ApiService {
       });
   }
 
+  /**
+   * Log the user out by reseting the token and remove user from local storage.
```
Comment (mhaswell-bcgov) [inline src/app/services/api.ts:115]:
```suggestion
   * Log the user out by resetting the token and remove user from local storage.
```

**Review by robrien-bcgov — APPROVED** [review]
looks like mostly just removed commented code, docblocks, and one removed import that wasn't being used in the input-modal component

File: src/app/services/api.ts [hunk 2]
```diff
@@ -517,7 +612,12 @@ export class ApiService {
     return this.http.get<Object>(`${this.pathAPI}/${queryString}`, {});
   }
 
-  // NB: returns array with 1 element
+  /**
+   * Get comment period. Get a CP with a specified set of fields.
```
Comment (mhaswell-bcgov) [inline src/app/services/api.ts:616]:
```suggestion
   * Get comment period. Get a comment period with a specified set of fields.
```
Seems to be referred to this way everywhere else

**Review by mhaswell-bcgov — APPROVED** [review]

File: src/app/services/api.ts (hunk 1 above)
Comment (Baelx) [reply src/app/services/api.ts:115]: Good catch! I'll update this in the follow-up PR

File: src/app/services/api.ts (hunk 2 above)
Comment (Baelx) [reply src/app/services/api.ts:616]: I'll revise the description here, it's a bit redundant too.

**Pull request merged**